scheduler/src/cook/mesos/scheduler.clj

```diff
@@ -590,7 +590,9 @@
   (for [{:keys [tasks leases]} matches
         :let [offers (mapv :offer leases)
               slave-id (-> offers first :slave-id :value)]
-        ^TaskAssignmentResult task tasks
+        ^TaskAssignmentResult task (->> tasks
+                                        ;; sort-by makes match transaction deterministic
+                                        (sort-by (comp :db/id :job #(.getRequest %))))
```

Comment:
At most there will be `max-considerable` tasks in the list. We currently run with 200 as `max-considerable` so the sort should be fairly lightweight.